include/ginkgo/core/distributed/matrix.hpp

```diff
@@ -508,6 +526,27 @@ class Matrix
                     ptr_param<const LinOp> local_matrix_template,
                     ptr_param<const LinOp> non_local_matrix_template);
 
+    explicit Matrix(std::shared_ptr<const Executor> exec,
```

Comment:
This could use a docstring. Also, what is the use case for this constructor, since it will leave non_local_matrix_ initialized? Is it for testing purposes only?

Reply:
It's for testing currently.

Reply:
I also use it to generate the local matrix-only distributed op in distributed pgm